src/server/cmd/pachctl/cmd/cmd.go

```diff
@@ -638,54 +639,47 @@ This resets the cluster to its initial state.`,
 	subcommands = append(subcommands, cmdutil.CreateAlias(portForward, "port-forward"))
 
 	var install bool
-	var path string
-	completion := &cobra.Command{
+	var installPathBash string
```

Comment:
You don't really need separate variables for `installPathBash` and `installPathZsh` as only one will be used in a run of `pachctl` - I'd recommend just reusing it as `installPath`.

Reply:
That didn't work because their default values are different, and cobra can't reuse the same variable name with different defaults (it'll end up using the last-defined default for all subcommands.)

The default in zsh is an empty string because I'm not aware of a reasonable value (unlike bash, which on linux is usually `/etc/bash-completions.d`.)

Reply:
Are you sure about the default value thing?  I don't think it should come into play unless cobra is parsing arguments for that specific command.

Reply:
It doesn't have to parse the variable, it's just following the pointer passed in

Reply:
You're right, that's pretty dumb of cobra =(